src/query/catalog/src/table_args.rs

```diff
@@ -12,6 +12,62 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
+use std::collections::HashMap;
+
+use common_exception::ErrorCode;
+use common_exception::Result;
 use common_expression::Scalar;
 
-pub type TableArgs = Option<Vec<Scalar>>;
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
+pub struct TableArgs {
+    pub positioned: Vec<Scalar>,
+    pub named: HashMap<String, Scalar>,
+}
+
+impl TableArgs {
+    pub fn is_empty(&self) -> bool {
+        self.named.is_empty() && self.positioned.is_empty()
+    }
+
+    pub fn new_positioned(args: Vec<Scalar>) -> Self {
+        Self {
+            positioned: args,
+            named: HashMap::new(),
+        }
+    }
+
+    pub fn expect_all_positioned(
```

Comment:
Add more comments for this function? From the name I can not get what its usage is.

Reply:
Others LGTM 👍

Reply:
done